modules/apps/depot/depot-service/src/main/java/com/liferay/depot/service/impl/DepotEntryLocalServiceImpl.java

```diff
@@ -142,6 +142,41 @@ public DepotEntry addDepotEntry(
 		return depotEntry;
 	}
 
+	@Override
+	public DepotEntry addOrUpdateDepotEntry(
+			String externalReferenceCode, Map<Locale, String> nameMap,
+			Map<Locale, String> descriptionMap,
+			Map<String, Boolean> depotAppCustomizationMap,
+			UnicodeProperties typeSettingsUnicodeProperties,
+			ServiceContext serviceContext)
+		throws PortalException {
+
+		DepotEntry depotEntry = null;
+
+		Group group = _groupLocalService.fetchGroupByExternalReferenceCode(
+			externalReferenceCode, serviceContext.getCompanyId());
+
+		if (group != null) {
+			depotEntry = getGroupDepotEntry(group.getGroupId());
+
+			depotEntry = updateDepotEntry(
+				depotEntry.getDepotEntryId(), nameMap, descriptionMap,
+				depotAppCustomizationMap, typeSettingsUnicodeProperties,
+				serviceContext);
+		}
+		else {
+			depotEntry = addDepotEntry(nameMap, descriptionMap, serviceContext);
+		}
+
+		group = depotEntry.getGroup();
+
+		group.setExternalReferenceCode(externalReferenceCode);
+
+		_groupLocalService.updateGroup(group);
+
+		return getDepotEntry(depotEntry.getDepotEntryId());
```

Comment:
Do we need to retrieve the depotEntry again? We already have it when we added it or updated it